pkg/templates/untrustedimages/template.go

```diff
@@ -0,0 +1,55 @@
+package untrustedimages
+
+import (
+	"fmt"
+	"regexp"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/untrustedimages/internal/params"
+	"golang.stackrox.io/kube-linter/pkg/templates/util"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	templateKey = "untrusted-images"
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Untrusted images",
+		Key:         templateKey,
+		Description: "Flag applications running containers with untrusted container images",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(p params.Params) (check.Func, error) {
+			return util.PerContainerCheck(func(container *v1.Container) []diagnostic.Diagnostic {
+
+				// checking whether the container image is a trusted or not
+
+				allowedResult, blockedResult := false, true
+
+				for _, allowed := range p.AllowedList {
+					allowedRe := regexp.MustCompile(allowed)
+					allowedResult = allowedRe.MatchString(container.Image)
```

Comment:
This seems wrong. You would want to `||` this with the current `allowedResult`, right?

Reply:
Sorry, I'm unable to understand this part.

Are you pointing [at this block](https://github.com/stackrox/kube-linter/blob/ddb195fe7855bb7c1482e5d94c2771accdd3ec66/pkg/templates/untrustedimages/template.go#L48-L50)?

```
if blockedResult || !allowedResult {...}
``` 
And could you please explain what is wrong. I'll try to correct it accordingly. Thank you!